attributes/default.rb

```diff
@@ -7,6 +7,8 @@
 # Init system to use
 default['mesos']['init']    = case node['platform']
                               when 'debian' then 'sysvinit_debian'
+                              when 'centos', 'amazon', 'scientific'
+                                node['platform_version'].to_i >= 7 ? 'systemd' : 'upstart'
```

Comment:
Hi @ovaistariq, in RHEL 6 sysvinit has been replaced by upstart. Please see [RHEL 6 Technical Notes](https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/6/html/6.0_Technical_Notes/deployment.html). Versions lower than 6 are not supported by this cookbook.